### PR TITLE
A few improvements

### PR DESCRIPTION
**Pull request by Makvagabo**
This is my proposal for some improvements in this project. Sorry for the big PR, it is not disrespect, it is how it was going.
I don't know whether it works or not in real a project, because I am just on a draft stage of my own project where I am gonna use ebubts. So check it please as soon as possible, I will be very appreciated.

- add prettier, formatting (I think it is a good popular tool for formatting code)
- turn on strict mode in TS (it is better to do this before it is too late)
- update tsconfig.json (I brought common and browser-focused settings)
- refactor chapterCleaner: use dompurify as a basement (I think now code looks more declarative and understandable, also we have more control)
- add tests for chapterCleaner (it is consequences and aproval of refactoring)
- fix ts errors
- rename some files and variable using snakeCase (it is common practice in JS community I suppose, I know it will be better adding some eslint for this, but lest do it in the next iterations)

### CODE REVIEW

File: src/chapterCleaner.ts [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+    const {
+      rawChapter,
+      manifest,
+      contentPath,
+      linkRoot = '',
+      imageRoot = '',
+    } = params;
+
+    DOMPurify.addHook('uponSanitizeElement', (currentNode, data) => {
```
Comment (Makvagabo) [inline src/chapterCleaner.ts:28]: I tried to repeat previous logic but it is still not clear for me, why we need this handling for src and href attributes.

File: tsconfig.json [hunk 2]
```diff
@@ -1,23 +1,23 @@
...
-        "./src/*"
-      ]
+      "*": ["./src/*"]
     }
   },
-  "include": [
-    "src/**/*"
-  ],
-  "exclude": [
-    "node_modules"
```
Comment (Makvagabo) [inline tsconfig.json:21]: it is default value

**Comment by GevatterGaul** [comment]
> This is my proposal for some improvements in this project. Sorry for the big PR, it is not disrespect, it is how it was going. I don't know whether it works or not in real a project, because I am just on a draft stage of my own project where I am gonna use ebubts. So check it please as soon as possible, I will be very appreciated.

No worries, your contribution is highly appreciated.
 
> * add prettier, formatting (I think it is a good popular tool for formatting code)

I have mixed experiences regarding prettier, mostly because of logistics. I don't mind settling on its opinionated style, but enforcing it is tedious. You need to enforce it via a pipeline but we don't have a public one. I use a private gitlab as second origin where I push the code to and run the pipeline steps. But I don't want that pipeline to make commits to github and no one can see the results.
I also think that in such a small project, we could get away without using it.

So, how about this: we add the package, so anyone who wants to use prettier can easily configure it in their VSCode or whatever. But we don't enforce the usage. I also will do a separate initial commit with the prettification, to keep it out of the commits concerning logic etc...

> * turn on strict mode in TS (it is better to do this before it is too late)

I had some issues with that when using xml2js. That's why it was deactivated. But if we replace xml2js anyway, we should enforce strict mode.

> * update tsconfig.json (I brought common and browser-focused settings)

I will check it out.

> * refactor chapterCleaner: use dompurify as a basement (I think now code looks more declarative and understandable, also we have more control)

What's the intention? Replace xml2js?

> * add tests for chapterCleaner (it is consequences and aproval of refactoring)

More tests is always good.

> * rename some files and variable using snakeCase (it is common practice in JS community I suppose, I know it will be better adding some eslint for this, but lest do it in the next iterations)

Isn't the style you want't to use called camelCase? I'm not aware of a common guidance here. Everyone seems to use something else. Google even recommends snake_case. I would not wanna use that ;-) . So, yes, we can use camelCase if you want for the file names. I'm not really opinionated on any of the styles, as long as we stay consistent.

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -14,20 +14,16 @@ default:
...
     policy: pull-push
 
-
-
 build:
   stage: build
   image: node:22.11.0-alpine3.20
   script:
-      - yarn install --dev --frozen-lockfile
-      - yarn build
```
Comment (GevatterGaul) [inline .gitlab-ci.yml:24]: I need that step. Not building the library in the pipeline was one of the reasons why the dist folder was missing in 0.1.4

File: .gitlab-ci.yml [hunk 4]
```diff
@@ -14,20 +14,16 @@ default:
...
+    - yarn install --dev --frozen-lockfile
 
 lint:
   stage: test
   image: node:22.11.0-alpine3.20
   script:
-    - yarn install --dev --frozen-lockfile
```
Comment (GevatterGaul) [inline .gitlab-ci.yml:30]: Kinda need that step as well. The lint command depends on it.

File: .gitlab-ci.yml [hunk 5]
```diff
@@ -43,7 +39,6 @@ test:
...
```
Comment (GevatterGaul) [inline .gitlab-ci.yml:46]: Same here

File: .prettierrc [hunk 6]
```diff
@@ -0,0 +1,3 @@
+{
+  "singleQuote": true
```
Comment (GevatterGaul) [inline .prettierrc:2]: Yes please.

File: CHANGELOG.md [hunk 7]
```diff
@@ -17,18 +17,18 @@
 
 ## 0.1.3
 
-* Fix tests
-* Update dependencies
+- Fix tests
```
Comment (GevatterGaul) [inline CHANGELOG.md:20]: Ehh, ok. This seems very opinionated. Is that a prettier thing?

File: package.json [hunk 8]
```diff
@@ -39,16 +40,20 @@
...
     "@types/xml2js": "^0.4.14",
     "eslint": "^9.8.0",
+    "eslint-config-prettier": "^9.1.0",
     "eslint-formatter-junit": "^8.40.0",
     "jsdom": "^25.0.1",
+    "prettier": "^3.4.2",
     "typescript": "^5.5.4",
     "typescript-eslint": "^8.0.1",
-    "vitest": "^2.0.5"
+    "vitest": "^2.1.8"
```
Comment (GevatterGaul) [inline package.json:49]: Ok, but shouldn't that be enforced by the yarn lock file?

File: src/chapterCleaner.test.ts [hunk 9]
```diff
@@ -0,0 +1,325 @@
+// noinspection ALL
```
Comment (GevatterGaul) [inline src/chapterCleaner.test.ts:1]: Could you do a separate commit with the renaming and then the changes? Should be way easier to see the diff then.

File: tsconfig.json (hunk 2 above)
Comment (GevatterGaul) [reply tsconfig.json:21]: The default values include even more paths: https://www.typescriptlang.org/tsconfig/#include . On the other hand: these have no impact on our current file structure. So, whatever...

File: src/epub.test.ts [hunk 10]
```diff
@@ -1,14 +1,19 @@
...
   let epubFile: File;
 
   beforeAll(() => {
-    const epubFileData = readFileSync('./Henry James - The Death of the Lion.epub');
-    epubFile = new File([epubFileData.buffer], 'Henry James - The Death of the Lion.epub');
+    const epubFileData = readFileSync(
+      './Henry James - The Death of the Lion.epub',
+    );
+    epubFile = new File(
+      [new Uint8Array(epubFileData)],
```
Comment (GevatterGaul) [inline src/epub.test.ts:14]: The conversion to Uint8Array should not be necessary. Please check wether your File type is from lib.dom.d.ts

**Review by GevatterGaul — CHANGES_REQUESTED** [review]
Ok, not donw yet with the chaptercleaner changes. But please have a look at the rest of the comments. I will do the prettification in the meantime to shrink this PR a bit.

**Comment by GevatterGaul** [comment]
Ok, I added prettier. Please merge the current master.

**Comment by Makvagabo** [comment]
> I have mixed experiences regarding prettier, mostly because of logistics. I don't mind settling on its opinionated style, but enforcing it is tedious. You need to enforce it via a pipeline but we don't have a public one. I use a private gitlab as second origin where I push the code to and run the pipeline steps. But I don't want that pipeline to make commits to github and no one can see the results.
I also think that in such a small project, we could get away without using it.
So, how about this: we add the package, so anyone who wants to use prettier can easily configure it in their VSCode or whatever. But we don't enforce the usage. I also will do a separate initial commit with the prettification, to keep it out of the commits concerning logic etc...

Ok, this is a kind of sensitive zone for everyone. I will drop Prettier.

> I had some issues with that when using xml2js. That's why it was deactivated. But if we replace xml2js anyway, we should enforce strict mode.

There is no any issues with xml2js, I fixed them all. What kind of issues did you encounter?

> What's the intention? Replace xml2js?

Roughly speaking because it is more preferable to use popular tools instead of self-written functions. We don't need to test tools like Dompurify, it is more safety as well - sanitize function does its work perfectly. No, I did touch xml2js at all. I just replace regexp functions.

> Isn't the style you want't to use called camelCase? I'm not aware of a common guidance here. Everyone seems to use something else. Google even recommends snake_case. I would not wanna use that ;-) . So, yes, we can use camelCase if you want for the file names. I'm not really opinionated on any of the styles, as long as we stay consistent.

Ok, I can rollback all this changes.

File: CHANGELOG.md (hunk 7 above)
Comment (Makvagabo) [reply CHANGELOG.md:20]: This is result of resolving conflicts. I did pay attention for that properly.

File: package.json (hunk 8 above)
Comment (Makvagabo) [reply package.json:49]: It was just my preference to see that vitest has latest version. I can roll it back.

File: src/epub.test.ts (hunk 10 above)
Comment (Makvagabo) [reply src/epub.test.ts:14]: The problem is that TS says that it needed to convert.
image

File: src/chapterCleaner.ts (hunk 1 above)
Comment (Makvagabo) [reply src/chapterCleaner.ts:28]: Could you explain to me why we need to reform href and src in that way?

File: .gitlab-ci.yml (hunk 3 above)
Comment (Makvagabo) [reply .gitlab-ci.yml:24]: I think it is a result of resolving conflicts. Sorry. Actually I did not touch this file. There was no reason to do this)

**Comment by Makvagabo** [comment]
Man, I realized that it is really a big PR, and I am gonna close this. It is difficult to make a review. I will bring my improvements separately by small PRs, and we will able to continue our conversation there.

File: package.json (hunk 8 above)
Comment (GevatterGaul) [reply package.json:49]: No, don't. I just wanted to understand why.

**Comment by GevatterGaul** [comment]
> > I have mixed experiences regarding prettier, mostly because of logistics. I don't mind settling on its opinionated style, but enforcing it is tedious. You need to enforce it via a pipeline but we don't have a public one. I use a private gitlab as second origin where I push the code to and run the pipeline steps. But I don't want that pipeline to make commits to github and no one can see the results.
> > I also think that in such a small project, we could get away without using it.
> > So, how about this: we add the package, so anyone who wants to use prettier can easily configure it in their VSCode or whatever. But we don't enforce the usage. I also will do a separate initial commit with the prettification, to keep it out of the commits concerning logic etc...
> 
> Ok, this is a kind of sensitive zone for everyone.

All good, I hope we found a good compromise.

> > I had some issues with that when using xml2js. That's why it was deactivated. But if we replace xml2js anyway, we should enforce strict mode.
> 
> There is no any issues with xml2js, I fixed them all. What kind of issues did you encounter?

Typing...

> > What's the intention? Replace xml2js?
> 
> Roughly speaking because it is more preferable to use popular tools instead of self-written functions. We don't need to test tools like Dompurify, it is more safety as well - sanitize function does its work perfectly. No, I did touch xml2js at all. I just replace regexp functions.

Ok, I will just check out the new PR.

> > Isn't the style you want't to use called camelCase? I'm not aware of a common guidance here. Everyone seems to use something else. Google even recommends snake_case. I would not wanna use that ;-) . So, yes, we can use camelCase if you want for the file names. I'm not really opinionated on any of the styles, as long as we stay consistent.
> 
> Ok, I can rollback all this changes.

Uah, no. I'm fine with camelCase. Just don't use this_style.

**Comment by GevatterGaul** [comment]
> Man, I realized that it is really a big PR, and I am gonna close this. It is difficult to make a review. I will bring my improvements separately by small PRs, and we will able to continue our conversation there.

Ok, go for it.

File: package.json (hunk 8 above)
Comment (Makvagabo) [reply package.json:49]: Just to be sure that we use the latest version, it is kind of paranoia.
